github/copilot.go

```diff
@@ -0,0 +1,284 @@
+		} else {
+			return fmt.Errorf("unsupported assignee type %v", v["type"])
+		}
+	default:
+		return fmt.Errorf("unsupported assignee type %T", v)
+	}
+
+	return nil
+}
+
+// GetUser gets the User from the CopilotSeatDetails if the assignee is a user.
+func (cp *CopilotSeatDetails) GetUser() (*User, bool) {
+	if user, ok := cp.Assignee.(*User); ok {
+		return user, true
+	}
+	return nil, false
+}
+
+// GetTeam gets the Team from the CopilotSeatDetails if the assignee is a team.
+func (cp *CopilotSeatDetails) GetTeam() (*Team, bool) {
+	if team, ok := cp.Assignee.(*Team); ok {
+		return team, true
+	}
+	return nil, false
+}
+
+// GetOrganization gets the Organization from the CopilotSeatDetails if the assignee is an organization.
+func (cp *CopilotSeatDetails) GetOrganization() (*Organization, bool) {
+	if organization, ok := cp.Assignee.(*Organization); ok {
+		return organization, true
+	}
+	return nil, false
+}
+
+// GetCopilotBilling gets Copilot for Business billing information and settings for an organization.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/copilot-for-business#get-copilot-for-business-seat-information-and-settings-for-an-organization
+func (s *CopilotService) GetCopilotBilling(ctx context.Context, org string) (*CopilotOrganizationDetails, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/copilot/billing", org)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var copilotDetails *CopilotOrganizationDetails
+	resp, err := s.client.Do(ctx, req, &copilotDetails)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return copilotDetails, resp, nil
+}
+
+// ListCopilotSeats lists Copilot for Business seat assignments for an organization.
+//
+// To paginate through all seats, populate 'Page' with the number of the last page.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/copilot-for-business#list-all-copilot-for-business-seat-assignments-for-an-organization
+func (s *CopilotService) ListCopilotSeats(ctx context.Context, org string, opts *ListOptions) (*ListCopilotSeatsResponse, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/copilot/billing/seats", org)
+
+	req, err := s.client.NewRequest("GET", u, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var copilotSeats *ListCopilotSeatsResponse
+	resp, err := s.client.Do(ctx, req, &copilotSeats)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return copilotSeats, resp, nil
+}
+
+// AddCopilotTeams adds teams to the Copilot for Business subscription for an organization.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/copilot-for-business#add-teams-to-the-copilot-for-business-subscription-for-an-organization
+func (s *CopilotService) AddCopilotTeams(ctx context.Context, org string, teamNames []string) (*SeatAssignments, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/copilot/billing/selected_teams", org)
+
+	req, err := s.client.NewRequest("POST", u, teamNames)
```

Comment:
This isn't going to work, and I'll show you why in the unit tests below...

OK, I can't stand the suspense.  :smile:  The problem is that this does not match the official documentation.
The POST request needs to look like this:

```
{"selected_teams":["engteam1","engteam2","engteam3"]}
```

but what you've written is instead going to look like this:

```
["engteam1","engteam2","engteam3"]
```

So what we need is a temporary struct, something like this:

```go
body := struct {
  SelectedTeams []string `json:"selected_teams"`
}{
  SelectedTeams: teamNames,
}
req, err := s.client.NewRequest("POST", u, body)
```

Reply:
I thought so as well hence why I initially had a struct for this and for `selected_users`, but went with the review suggestion to just use a slice since I thought maybe I had overcomplicated things 😅

Reply:
No, I think you were correct, @WillAbides - we should be passing in a slice of strings, and let the endpoint method handle the formatting of the call internally.